jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -331,6 +333,45 @@ def _parse_spectrum1d(app, file_obj, data_label=None):
     app.add_data_to_viewer('spectrum-viewer', f"{data_label}[FLUX]")
 
 
+def _parse_ndarray(app, file_obj, data_label=None, data_type=None):
+    if data_label is None:
+        data_label = 'Unknown array'
+
+    if data_type is None:
+        data_type = 'flux'
+
+    flux = np.fliplr(np.rot90(np.moveaxis(file_obj, 0, 2), k=-1, axes=(0, 1)))
+    if not hasattr(flux, 'unit'):
+        flux = flux << u.count
+    s3d = Spectrum1D(flux=flux)
+    app.add_data(s3d, data_label)
+
+    if data_type == 'flux':
+        app.add_data_to_viewer('flux-viewer', data_label)
+        app.add_data_to_viewer('spectrum-viewer', data_label)
+    elif data_type == 'uncert':
+        app.add_data_to_viewer('uncert-viewer', data_label)
+
+
+def _parse_gif(app, file_obj, data_label=None):  # pragma: no cover
+    # NOTE: Parsing GIF needs imageio and Pillow, both are which undeclared
+    # in setup.cfg but might or might not be installed by declared ones.
+    import imageio
+
+    file_name = os.path.basename(file_obj)
+
+    if data_label is None:
+        data_label = f'{os.path.splitext(file_name)[0]}[FLUX]'
```

Comment:
It was late at night when I wrote this one but I vaguely remember `app.add_data` throwing out error about "extension not found" unless I add this `[FLUX]` to the label. 🤷